examples/simple/src/posts/PostCreate.js

```diff
@@ -18,24 +16,36 @@ import {
     SimpleFormIterator,
     TextInput,
     Toolbar,
-    crudCreate,
     required,
+    useCreate,
+    useRedirect,
 } from 'react-admin'; // eslint-disable-line import/no-unresolved
-
-const saveWithNote = (values, basePath, redirectTo) =>
-    crudCreate('posts', { ...values, average_note: 10 }, basePath, redirectTo);
+import { useForm } from 'react-final-form';
 
 const SaveWithNoteButton = props => {
-    const dispatch = useDispatch();
-    const { basePath, handleSubmit, redirect } = props;
+    const [create] = useCreate('posts');
+    const redirectTo = useRedirect();
+    const { basePath, redirect } = props;
 
-    const handleClick = useCallback(
-        () =>
-            handleSubmit(values => {
-                dispatch(saveWithNote(values, basePath, redirect));
-            }),
-        [basePath, dispatch, handleSubmit, redirect]
-    );
+    const form = useForm();
+    const handleClick = useCallback(() => {
+        const formState = form.getState();
+        if (!formState.valid) {
+            return;
+        }
+
+        create(
+            null,
+            {
+                data: { ...formState.values, average_note: 10 },
+            },
+            {
+                onSuccess: ({ data: newRecord }) => {
+                    redirectTo(redirect, basePath, newRecord.id, newRecord);
+                },
+            }
+        );
+    }, [create, form, redirect, redirectTo, basePath]);
```

Comment:
I'm afraid the `form` object never changes (only its properties), and `useCallback` does identity check. Can't this cause a bug? shouldn't you track the values instead?